homeassistant/components/mazda/button.py

```diff
@@ -0,0 +1,153 @@
+"""Platform for Mazda button integration."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import Any
+
+from pymazda import Client as MazdaAPIClient
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+
+async def handle_button_press(
+    client: MazdaAPIClient,
+    key: str,
+    vehicle_id: int,
+    coordinator: DataUpdateCoordinator,
+) -> None:
+    """Handle a press for a Mazda button entity."""
+    api_method = getattr(client, key)
+
+    try:
+        await api_method(vehicle_id)
+    except Exception as ex:
```

Comment:
Please limit exceptions to a tuple of expected exceptions instead of a broad exception catch